gremlin-go/README.md

```diff
@@ -117,47 +68,114 @@ require (
 ```
 As well as a populated `go.sum` file.
 
-This following example should run, provided that it is configured to point to a compatible `gremlin-server`. In this example, a simple local server is running on port 8182, and this will print`[2]` as an output. If no server is available, this code can still be executed to print an error as output.
+*if there are no usages for gremlingo found, it will remove the require from go.mod and not import dependencies.*
 
+## Simple usage
+
+Here is a simple example of using Gremlin-Go as an import in a sample project's `main.go` file. 
 ```go
 package main
 
 import (
-	"fmt"
 	"github.com/apache/tinkerpop/gremlin-go/driver"
 )
 
+func main() {
+	// Simple stub to use the import. See subsequent section for actual usage. 
+	_, _ = gremlingo.NewDriverRemoteConnection(fmt.Sprintf("ws://%s:%d/gremlin", "localhost", 8182))
+}
+```
+
+*Pay attention to the suffix `/gremlin` in connection string*
+
+This following example should run, provided that it is configured to point to a compatible `gremlin-server`. In this example, a simple local server is running on port 8182, and this will print `[0]` as an output. If no server is available, this code can still be executed to print an error as output.
+
+```go
+package main
+
+import (
+	"fmt"
+	"github.com/apache/tinkerpop/gremlin-go/v3/driver"
+)
+
 func main() {
 	// Creating the connection to the server.
-	driverRemoteConnection, err := gremlingo.NewDriverRemoteConnection(fmt.Sprintf("ws://%s:%d", "localhost", 8182))
+	driverRemoteConnection, err := gremlingo.NewDriverRemoteConnection("ws://localhost:8182/gremlin")
+	// Handle error
 	if err != nil {
 		fmt.Println(err)
 		return
 	}
 	// Cleanup
 	defer driverRemoteConnection.Close()
-	
-	// Submit a simple string traversal
-	resultSet, err := driverRemoteConnection.Submit("1 + 1")
+
+	// Creating graph traversal
+	g := gremlingo.Traversal_().WithRemote(driverRemoteConnection)
+
+	// Perform traversal
+	result, err := g.V().Count().ToList()
 	if err != nil {
 		fmt.Println(err)
 		return
 	}
-	
-	// Grab the first result from all the results in the ResultSet.
-	result := resultSet.All()[0]
-	
-	// Print the first result.
-	fmt.Println(result.GetString())
+	fmt.Println(result[0].GetString())
 }
 ```
 
 Note: The exact import name as well as the module prefix for `NewDriverRemoteConnection` may change in the future.
 
-### Design Architecture
+## Customizing connection
+`gremlingo.NewDriverRemoteConnection` accepts a config function as a parameter. (See code documentation for additional parameters and their usage) <!--TODO: when merged add link to https://github.com/lyndonbauto/tinkerpop/blob/valentyn/AN-988-reference/docs/src/reference/gremlin-variants.asciidoc#configuration-3 -->
```

Comment:
We can also add a link to the godocs at pkg.go

Reply:
https://pkg.go.dev/github.com/apache/tinkerpop/gremlin-go/v3 is just this README
https://pkg.go.dev/github.com/apache/tinkerpop/gremlin-go/v3/driver contains list of exported functions, can be useful

Reply:
updated in https://github.com/lyndonbauto/tinkerpop/pull/88/commits/31a3b30caf421ac03e42c24c68c768e7fa7bb193